enginetest/queries/table_func_scripts.go

```diff
@@ -170,8 +170,32 @@ var TableFunctionScriptTests = []ScriptTest{
 		Expected: []sql.Row{{0}, {1}, {2}, {3}, {4}},
 	},
 	{
+		Name:            "sequence_table allows point lookups",
 		Query:           "select * from sequence_table('x', 5) where x = 2",
 		Expected:        []sql.Row{{2}},
 		ExpectedIndexes: []string{"x"},
 	},
+	{
+		Name:            "sequence_table allows range lookups",
```

Comment:
these tests all look good, the code changes are for merge-join though. should we have a merge join test?

Reply:
I added a test that we don't try to generate merge joins when the index doesn't support ranges.

Reply:
Thanks, looks like i hit this error also just now in my GMS bump
```
TestHistorySystemTable/dolt_history_table_primary_key_with_join/dolt_history_table_primary_key_with_join/_SELECT___dolt_history_xyz.x_as_x,___dolt_history_xyz.y_as_y,___dolt_history_xyz.z_as_z,___dolt_commits.commit_hash_as_comm_FROM___dolt_history_xyz___LEFT_JOIN___dolt_commits___ON___dolt_history_xyz.commit_hash_=_dolt_commits.commit_hash_ORDER_BY___dolt_history_xyz.x,___dolt_history_xyz.y,___dolt_history_xyz.z; (0.00s)
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x1a3351d]

goroutine 297934 [running]:
testing.tRunner.func1.2({0x2e3a6c0, 0x6648520})
	/Users/runner/hostedtoolcache/go/1.21.0/x64/src/testing/testing.go:1545 +0x238
testing.tRunner.func1()
	/Users/runner/hostedtoolcache/go/1.21.0/x64/src/testing/testing.go:1548 +0x397
panic({0x2e3a6c0?, 0x6648520?})
	/Users/runner/hostedtoolcache/go/1.21.0/x64/src/runtime/panic.go:914 +0x21f
github.com/dolthub/go-mysql-server/sql/memo.(*Memo).MemoizeMergeJoin(0x0?, 0xc004e90180, 0xc000e51e80, 0xc000e51f80, 0x0, 0x0, 0x11, {0xc007702740, 0x1, 0x1}, ...)
	/Users/runner/go/pkg/mod/github.com/dolthub/go-mysql-server@v0.17.1-0.20230905161304-74cfc27aece4/sql/memo/memo.go:448 +0x17d
github.com/dolthub/go-mysql-server/sql/analyzer.addMergeJoins.func1({0x3cb3050?, 0xc00297a748})
```